src/main/java/ser/mil/rideapp/domain/service/RideService.java

```diff
@@ -21,20 +29,32 @@ public void orderRide(String from, String to, String customer) {
         rideRepository.save(new Ride("1",from, to, customer, price, RideStatus.PENDING));
     }
 
-    public void pairPassengerWithDriver(){
-        Ride pendingRide= rideRepository.getRides().stream()
-                .filter(ride -> ride.getStatus().equals(RideStatus.PENDING))
-                .findFirst()
-                .orElse(null);
-        Driver availableDriver = rideRepository.getDrivers().stream()
-                .filter(Driver::getAvailable).findFirst().orElse(null);
-        if (pendingRide != null) {
+    public void pairPassengerWithDriver() {
+        logger.debug("Rozpoczęcie procesu parowania pasażera z kierowcą...");
+
+        List<Ride> rides = rideRepository.pendingRides();
+        List<Driver> drivers = rideRepository.getDrivers().stream()
+                .filter(Driver::getAvailable)
+                .collect(Collectors.toList());
+
+        while (!rides.isEmpty() && !drivers.isEmpty()) {
+            Ride pendingRide = rides.removeFirst();  // Pobierz i usuń pierwszy przejazd
+            Driver availableDriver = drivers.removeFirst(); // Pobierz i usuń pierwszego dostępnego kierowcę
+
             pendingRide.setDriver(availableDriver);
-        }
-        if (pendingRide != null) {
             pendingRide.setStatus(RideStatus.FOUND);
+            availableDriver.setAvailable(false);
+
+            logger.info("Pasażer {} został sparowany z kierowcą {}",
+                    pendingRide.getCustomer(), availableDriver.getFirstName());
         }
 
-        System.out.println(pendingRide);
+        if (rides.isEmpty()) {
+            logger.debug("Brak oczekujących przejazdów do sparowania.");
+        }
+        if (drivers.isEmpty()) {
+            logger.debug("Brak dostępnych kierowców.");
+        }
```

Comment:
Zamiast tych dwóch logowań zrób jedno logowanie (po pętli while) który pisze że zakończono parowanie i wypisze ile zostało pending rides i available drivers. To w logach nam będzie dużo dawało